akka-diagnostics/src/test/java/akka/diagnostics/ConfigCheckerSpec.scala

```diff
@@ -0,0 +1,1087 @@
+        """akka.cluster.down-removal-margin = 10s
+          akka.cluster.split-brain-resolver.stable-after = 15s
+          akka.cluster.split-brain-resolver.active-strategy = keep-majority""",
+        s"""
+          akka.cluster.auto-down-unreachable-after = 10s
+          akka.cluster.split-brain-resolver.active-strategy = keep-majority
+          akka.diagnostics.checker.disabled-checks = [auto-down, typo]
+          """)
+
+      val configs = configStrings.map(c =>
+        ConfigFactory
+          .parseString(c)
+          .withFallback(defaultCluster))
+      configs.zipWithIndex.foreach { case (c, i) =>
+        withClue(s"problem with config #${i + 1}") {
+          val checker = new ConfigChecker(extSys, c, reference)
+          val warnings = checker.check().warnings
+          printDocWarnings(warnings)
+          assertCheckerKey(warnings, "split-brain-resolver")
+          assertDisabled(c, "split-brain-resolver")
+        }
+      }
+    }
+
+    "not check SBR if reference.conf doesn't contain split-brain-resolver section" in {
+      // akka-split-brain-resolver is in classpath and in this test we want to simulate that it wasn't
+      val referenceWithoutSbr = ConfigFactory.defaultReference().withoutPath("akka.cluster.split-brain-resolver")
+      val clusterWithoutSbr = defaultCluster.withoutPath("akka.cluster.split-brain-resolver")
+
+      val checker = new ConfigChecker(extSys, clusterWithoutSbr, referenceWithoutSbr)
+      val warnings = checker.check().warnings
+      warnings should ===(Nil)
+    }
+
+    "not check SBR if reference.conf doesn't contain split-brain-resolver section, even if included in application.conf" in {
+      // akka-split-brain-resolver is in classpath and in this test we want to simulate that it wasn't
+      val referenceWithoutSbr = ConfigFactory.defaultReference().withoutPath("akka.cluster.split-brain-resolver")
+      val clusterWithoutSbr = defaultCluster.withoutPath("akka.cluster.split-brain-resolver")
+      val c = ConfigFactory
+        .parseString("""
+        akka.cluster.split-brain-resolver.active-strategy = keep-majority
+        akka.cluster.down-removal-margin = 1s
+        """)
+        .withFallback(clusterWithoutSbr)
+
+      val checker = new ConfigChecker(extSys, c, referenceWithoutSbr)
+      val warnings = checker.check().warnings
+      // it's reported as a typo since not included in reference.conf (as expected)
+      assertCheckerKey(warnings, "typo")
+    }
+
+    "log warning when ActorSystem startup" in {
+      val c = ConfigFactory
+        .parseString("""
+          akka.log-level = INFO # typo, it should be akka.loglevel
+          akka.diagnostics.checker.async-check-after = 200ms
+        """)
+        .withFallback(system.settings.config)
+      val logSource = classOf[ConfigChecker].getName
+      // the logging is performed async after 200ms, and therefore we can intercept the log like this
+      val sys2 = ActorSystem(system.name + "-2", c)
```

Comment:
I can't find how this triggers the check of the config checker. I think that bit is missing here. What do you think @patriknw

Reply:
It could be that you need to add `ConfigChecker.reportIssues(system)` in `DiagnosticsExtension`.
That extension is loaded when the actor system is started and it should trigger a config check.